main.go

```diff
@@ -52,13 +52,14 @@ type genOptions struct {
 	NoRecurse bool         `long:"no-recurse" description:"Don't generate code for included Thrift files."`
 	Plugins   plugin.Flags `long:"plugin" short:"p" value-name:"PLUGIN" description:"Code generation plugin for ThriftRW. This option may be provided multiple times to apply multiple plugins."`
 
-	GeneratePluginAPI bool `long:"generate-plugin-api" hidden:"true" description:"Generates code for the plugin API"`
-	NoVersionCheck    bool `long:"no-version-check" hidden:"true" description:"Does not add library version checks to generated code."`
-	NoTypes           bool `long:"no-types" description:"Do not generate code for types, implies --no-service-helpers."`
-	NoConstants       bool `long:"no-constants" description:"Do not generate code for const declarations."`
-	NoServiceHelpers  bool `long:"no-service-helpers" description:"Do not generate service helpers."`
-	NoEmbedIDL        bool `long:"no-embed-idl" description:"Do not embed IDLs into the generated code."`
-	NoZap             bool `long:"no-zap" description:"Do not generate code for Zap logging."`
+	GeneratePluginAPI bool   `long:"generate-plugin-api" hidden:"true" description:"Generates code for the plugin API"`
+	NoVersionCheck    bool   `long:"no-version-check" hidden:"true" description:"Does not add library version checks to generated code."`
+	NoTypes           bool   `long:"no-types" description:"Do not generate code for types, implies --no-service-helpers."`
+	NoConstants       bool   `long:"no-constants" description:"Do not generate code for const declarations."`
+	NoServiceHelpers  bool   `long:"no-service-helpers" description:"Do not generate service helpers."`
+	NoEmbedIDL        bool   `long:"no-embed-idl" description:"Do not embed IDLs into the generated code."`
+	NoZap             bool   `long:"no-zap" description:"Do not generate code for Zap logging."`
+	SingleFile        string `long:"single-file" value-name:"FILENAME" description:"Generates a single .go file as an output."`
```

Comment:
OutputFile?

Also, we'll need some sort of check somewhere to ensure that you haven't provided a filename if multiple Thrift files are involved because the name won't apply to included files.

Reply:
Changed the name. I don't believe this check is needed. From my reading of the code I don't think multiple thrift files can be specified in an input.

Reply:
Multiple Thrift files can't be specified as input but without the `--no-recurse` flag, if a file includes another Thrift file, it will be generated too.

Reply:
If this behaviour is not allowed I think it would be safe to mandate --no-recurse if the flag is specified. It seems like restricting and documenting the behaviour is safer, what are your thoughts?

Reply:
Yeah, I'm okay with saying "--output-file" implies "--no-recurse"

Reply:
Okay, i've made the change.